test_report/tests/report.rs

```diff
@@ -0,0 +1,97 @@
+use prost::Message;
+use prost_wkt_types::Timestamp;
+use proto::test_context::test_run::TestCaseRunStatus;
+use tempfile::tempdir;
+use test_report::report::{MutTestReport, Status};
+use test_utils::mock_git_repo::setup_repo_with_commit;
+use test_utils::mock_server::MockServerBuilder;
+
+#[tokio::test(flavor = "multi_thread")]
+async fn publish_test_report() {
+    let temp_dir = tempdir().unwrap();
+    let repo_setup_res = setup_repo_with_commit(&temp_dir);
+    assert!(repo_setup_res.is_ok());
+    let set_current_dir_res = std::env::set_current_dir(&temp_dir);
+    assert!(set_current_dir_res.is_ok());
```

Comment:
nit: asserting on `is_ok` won't give a track as to what went wrong—I think it's preferred to `unwrap()` in tests for this purpose